include/vctr/Miscellaneous/JUCEHelpers.h

```diff
@@ -0,0 +1,107 @@
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+namespace vctr::juce_helpers
+{
+
+/** Loads the first elements of src into a juce::SIMDRegister.
+
+    Note: Only available if VCTR is used in a JUCE based project that uses the
+    juce_dsp module.
+
+    This function expects that the number of elements in src is exactly equal
+    to juce::SIMDRegister<Src::value_type>::SIMDNumElements. Note that JUCE
+    chooses either AVX or SSE registers on x64 platforms, depending on the
+    AVX flags being passed to the compiler. In particular, this means that
+    the function could expect an 8 element float vector as source when built
+    with AVX flags while expecting a 4 element vector otherwise. So cross platform
+    compatibility is best when not building with AVX flags.
+
+    If you want to be more explicit about the SIMD register type, consider using
+    the explicit SIMD related member functions of VctrBase.
+
+    @see VctrBase::getNeon
+    @see VctrBase::getSSE
+    @see VctrBase::getAVX
+ */
+template <is::anyVctr Src>
+auto toSIMDRegister (const Src& src)
+{
+    using SIMDRegister = juce::dsp::SIMDRegister<typename Src::value_type>;
+
+    VCTR_ASSERT (src.size() == SIMDRegister::SIMDNumElements);
+
+    if constexpr (Config::archARM)
+    {
+        return SIMDRegister { src.getNeon (0).value };
+    }
+    else
+    {
+        constexpr size_t registerSize = SIMDRegister::SIMDRegisterSize;
+
+        if constexpr (registerSize == 32)
+        {
+            return SIMDRegister { src.getAVX (0).value };
+        }
+
+        return SIMDRegister { src.getSSE (0).value };
+    }
+}
+
+/** Returns a Span that views a single channel of a juce::dsp::AudioBlock.
+
+    This allows you to operate on audio channels with all available VCTR features.
+    In case the SampleType of the block is const, the Span will give you read access,
+    otherwise it will also allow you write access to the channel data.
+
+    Note: If you are working with audio blocks that view memory allocated by JUCE,
+    the allocated memory should be SIMD aligned. In that case consider using
+    blockChannelAsSpanSIMDAligned, which will return a Span that expects the memory
+    being SIMD aligned and might chose some better optimised versions of VCTR expressions
+    without any runtime overhead. However, if you are accessing the audio buffer passed
+    from a plugin host you cannot expect any guaranteed alignment. In this case, you should
+    use this function.
+
+    @see blockChannelAsSpanSIMDAligned
+ */
+template <class SampleType>
+auto blockChannelAsSpan (const juce::dsp::AudioBlock<SampleType>& block, size_t channelIdx)
+{
+    auto numSamples = block.getNumSamples();
+    return Span { block.getChannelPointer (channelIdx), numSamples };
```

Comment:
should this maybe assert that the channel index is within the number of available channels?

Reply:
`getChannelPointer` already should assert that internally so I don't see a point in catching that twice